### PR TITLE
Refactoring to single serverConnectorControler architecture

### PR DESCRIPTION
**Pull request by Yasara123**
1) Refactoring to single serverConnectorControler architecture for performance.
2) Change header value separating regex
3) Reformatting.

### CODE REVIEW

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 1]
```diff
@@ -27,25 +27,144 @@
...
 
-import java.util.Collections;
-import java.util.HashSet;
 import java.util.Map;
 import java.util.concurrent.ConcurrentHashMap;
 
 /**
  * {@code HttpConnectorRegistry} The code is responsible for maintaining the all active server connectors.
  */
 class HttpConnectorRegistry {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: using package private modifier intentionally?

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 2]
```diff
@@ -27,25 +27,144 @@
...
+     * Register new source listener.
+     *
+     * @param sourceEventListener the source event listener.
+     * @param listenerUrl the listener url.
+     * @param workerThread the worker thread count of siddhi level thread pool executor.
+     * @param isAuth the authentication is required for source listener.
+     */
+    void registerSourceListener(SourceEventListener sourceEventListener, String listenerUrl, int
+            workerThread, Boolean isAuth) {
+        synchronized (this.sourceListenersMap) {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: why are you synchronizing on a concurrent map object

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 3]
```diff
@@ -27,25 +27,144 @@
...
+            if (this.sourceListenersMap.get(HttpSourceUtil.getSourceListenerKey(listenerUrl)) != null) {
+                throw new ExecutionPlanCreationException("Listener URL " + listenerUrl + " already connected.");
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: HttpSourceUtil.getSourceListenerKey(listenerUrl) is called twice. use local variables and make only one call

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 4]
```diff
@@ -27,25 +27,144 @@
...
+        }
+    }
+
+    /**
+     * Unregister the source listener.
+     *
+     * @param listenerUrl the listener url
+     */
+    void unregisterSourceListener(String listenerUrl) {
+        synchronized (this.sourceListenersMap) {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: why are you synchronizing on a concurrent map object

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 5]
```diff
@@ -27,25 +27,144 @@
...
+            String key = HttpSourceUtil.getSourceListenerKey(listenerUrl);
+            HttpSourceListener httpSourceListener = this.sourceListenersMap.get(key);
+            if (httpSourceListener != null) {
+                httpSourceListener.disconnect();
+                this.sourceListenersMap.remove(key);
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: remove will return you the object anyway. So no need to use a get

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 6]
```diff
@@ -27,25 +27,144 @@
...
+    /**
+     * Register the new server connector.
+     *
+     * @param listenerUrl the listener url
+     */
+    void unregisterServerConnector(String listenerUrl) {
+        Boolean isContainedAnotherDependentListener = false;
+        String port = HttpSourceUtil.getPort(listenerUrl);
+        String listenerKey = HttpSourceUtil.getSourceListenerKey(listenerUrl);
+        synchronized (this.serverConnectorMap) {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: why are you synchronizing on a concurrent map object

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 7]
```diff
@@ -27,25 +27,144 @@
...
+    }
+
+    /**
+     * Check whether that particular source listener is there.
+     *
+     * @param key unique key (port-context) of the source listener
+     * @return contain or not
+     */
+    boolean isListenerContain(String key) {
+        return this.sourceListenersMap.containsKey(key);
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: this method is not needed. move this to the caller method itself

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java [hunk 8]
```diff
@@ -27,25 +27,144 @@
...
     }
 
+    /**
+     * Stop server connector controller.
+     */
+    void stopHttpServerConnectorController() {
+        synchronized (this.sourceListenersMap) {
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: synchronization issue

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpSource.java [hunk 9]
```diff
@@ -154,77 +151,64 @@
...
-                                listenerUrl, isAuth, sourceEventListener));
-            }
-        }
-    }
-
-    static Map<String, HttpSourceListener> getRegisteredSourceListenersMap() {
-        return registeredSourceListenersMap;
+        this.listenerConfig = new HttpSourceUtil().setListenerProperty(this.listenerUrl, configReader);
+        HttpConnectorRegistry.getInstance().initHttpServerConnector(configReader);
+        HttpConnectorRegistry.getInstance().registerSourceListener(sourceEventListener, this.listenerUrl,
```
Comment (tishan89) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpSource.java]: get the instance in init, store locally and reuse

File: component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java (hunk 1 above)
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/source/HttpConnectorRegistry.java]: it is intentionally.

File: component/src/main/java/org/wso2/extension/siddhi/io/http/sink/util/HttpSinkUtil.java [hunk 10]
```diff
@@ -83,16 +83,18 @@ public HttpSinkUtil() {
      * @return list of headers.
      */
     public List<Header> getHeaders(String headers) {
+        headers = headers.trim();
+        headers = headers.substring(1, headers.length() - 1);
```
Comment (Yasara123) [inline component/src/main/java/org/wso2/extension/siddhi/io/http/sink/util/HttpSinkUtil.java]: Added example at java doc.  Fix in  c909244.

**Pull request merged**